cmd/log-parser/README.md

```diff
@@ -0,0 +1,94 @@
+# `kata-log-parser`
+
+* [Logfile requirements](#logfile-requirements)
+* [Component logfiles](#component-logfiles)
+* [Usage](#usage)
+
+`kata-log-parser` is a tool that combines logfiles generated by the various
+system components, sorts them by timestamp and re-displays the log entries,
+adding a time delta showing how much time has elapsed between each log entry.
+
+The tool also checks the validity of all log records and can re-format the
+logs and output them in a different format.
+
+For full details,
+
+```
+$ kata-log-parser --help
+```
+
+## Logfile requirements
+
+The tool reads logfiles in the [logfmt](https://brandur.org/logfmt) structured
+logging format, for example as created by the golang
+[logrus](https://godoc.org/github.com/sirupsen/logrus) package.
+
+The tool requires the following fields to be defined for each log record:
+
+- Log level field (`level`): must be one of the logrus `LogLevel` values
+  in string format (for example `debug`, `info`, `error`).
+
+- Process ID field (`pid`): the numeric process identifier for the process
+  that generated the log record.
+
+- Timestamp field (`time`): in [RFC3339](https://www.ietf.org/rfc/rfc3339.txt)
+  format and including a nanosecond value.
+
+- Source field (`source`): a single word that specifies the name of a unique
+  part of the system (for example, `proxy`, `runtime`, `shim`).
+
+- Name field (`name`): a single word that specifies the name of the
+  application that generated the log record (for example, `kata-runtime`).
+
+
+## Component logfiles
+
+The primary logfiles the tool reads are:
+
+- The [proxy](https://github.com/kata-containers/proxy) log.
+
+  This log also includes embedded log entries from the
+  [agent](https://github.com/katacontainers/agent). `kata-log-parser`
+  automatically unpacks and displays these, discarding the proxies encapsulating
```

Comment:
automatically unpacks and displays the entries. During this process, the encapsulating proxy log messages are discarded.